test/e2e/workflow/deployment_test.go

```diff
@@ -133,6 +133,57 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			components = append(components, MacvtapComponent)
 			testConfigUpdate(gvk, configSpec, components)
 		})
+		Context("and PlacementConfiguration is deployed", func() {
```

Comment:
Heh, I missed something. This new test is kind of redundant since it is already covered in the unit test.

What we should check here is deployment of some components. Deploy it with the default placement and then with an alternative placement, check that all the components are still ok and we turn available.

Can we add such a test? My main worry is that some of the components would fail to adapt to the new placement OR that our conditions would not be able to track removal of components with the old placement and deployment of the new ones.

Reply:
the motivation for adding this test was that the unit test didn't catch that PlacementConfiguration was actually broken on master branch until fixed [here](https://github.com/kubevirt/cluster-network-addons-operator/pull/640).

But we can do that also by adding it to the existing test deploying it with the other components.
Let me know what you think

Reply:
Do we want the change of placementConfiguration to immediately apply on components pods? I imagine that users may expect that.
Currently though, I think that updating placementConfiguration will only affect future component pods, and the existing will remain as before.

Reply:
@rhrazdil  We apply it immediately on all deployed components.  When the CR is changed, we reconcile it, render new objects for all components and apply them.

Reply:
@RamLavi why adding to an existing one? If this was indeed an issue, let's keep the test. However, I still miss a test showing that we are able to apply the changes on live components, turn "Progressing" and eventually become "Available" again.

Reply:
ack, will add more components to the original test..

Reply:
Done